internal/service/handlers/daily_question_edit.go

```diff
@@ -0,0 +1,197 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/rarimo/geo-auth-svc/pkg/auth"
+	"github.com/rarimo/geo-points-svc/internal/data"
+	"github.com/rarimo/geo-points-svc/internal/service/requests"
+	"github.com/rarimo/geo-points-svc/resources"
+	"gitlab.com/distributed_lab/ape"
+	"gitlab.com/distributed_lab/ape/problems"
+)
+
+func EditDailyQuestion(w http.ResponseWriter, r *http.Request) {
+	if !auth.Authenticates(UserClaims(r), auth.AdminGrant) {
+		ape.RenderErr(w, problems.Unauthorized())
+		return
+	}
+
+	IDStr := strings.ToLower(chi.URLParam(r, "question_id"))
+	ID, err := strconv.ParseInt(IDStr, 10, 64)
+	if err != nil {
+		Log(r).WithError(err).Error("Failed to parse ID")
+		ape.RenderErr(w, problems.BadRequest(err)...)
+		return
+	}
+
+	req, err := requests.NewDailyQuestionEdit(r)
+	if err != nil {
+		Log(r).WithError(err).Error("Error creating daily question edit request")
+		ape.RenderErr(w, problems.InternalError())
+		return
+	}
+
+	question, err := DailyQuestionsQ(r).FilterByID(ID).Get()
+	if err != nil {
+		Log(r).WithError(err).Error("Error getting question")
+		ape.RenderErr(w, problems.InternalError())
+		return
+	}
+	if question == nil {
+		Log(r).Warnf("Question with ID %d not found", ID)
+		ape.RenderErr(w, problems.NotFound())
+		return
+	}
+
+	nowTime := time.Now().UTC()
+	if !question.StartsAt.After(time.Date(nowTime.Year(), nowTime.Month(), nowTime.Day(), 0, 0, 0, 0, time.UTC)) {
+		Log(r).Warnf("Cannot change a question id: %v that is available today or in the past", ID)
+		ape.RenderErr(w, problems.Forbidden())
+		return
+	}
```

Comment:
You should check midnight of the next day, because admin can change question on the day in which users already answer to questions